app/models/user.rb

```diff
@@ -4,4 +4,30 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+
+  def has_membership_application?
+    membership_applications.size > 0 && (membership_applications.select { |app| app.persisted? }).count > 0
+  end
+
+
+  def has_company?
+    has_membership_application? && (membership_applications.select { |app| app.company && app.company.persisted? }).count > 0
+  end
+
+
+  def membership_application
+    has_membership_application? ? membership_applications.last : nil
+  end
```

Comment:
Thomas, I was referring to the convention that you implemented with your commit in PR #15 [line 24 in app/views/paritals/_top.html.haml](https://github.com/tochman/shf-project/blob/4a9b9f1dee24e101f7a709a0d9dfab2150903cdf/app/views/partials/_top.html.haml#L24)